derive/src/filter.rs

```diff
@@ -47,54 +47,10 @@ pub fn filter_struct(
 ) -> Result<TokenStream, crate::error::Error> {
     let fields: Vec<TokenStream> = fields
         .iter()
-        .map(|(ident, ty)| {
-            let type_literal = ty.to_token_stream().to_string();
-
-            let default_filters = vec![
-                "String",
-                "i8",
-                "i16",
-                "i32",
-                "i64",
-                "u8",
-                "u16",
-                "u32",
-                "u64",
-                "f32",
-                "f64",
-                #[cfg(feature = "with-chrono")]
-                "Date",
-                #[cfg(feature = "with-chrono")]
-                "DateTime",
-                #[cfg(feature = "with-chrono")]
-                "DateTimeUtc",
-                #[cfg(feature = "with-chrono")]
-                "DateTimeWithTimeZone",
-                #[cfg(feature = "with-decimal")]
-                "Decimal",
-                #[cfg(feature = "with-json")]
-                "Json",
-                #[cfg(feature = "with-uuid")]
-                "Uuid",
-                "BinaryVector",
-                "bool",
-            ];
-
-            let filter_item = if default_filters.contains(&type_literal.as_str())
-                || type_literal.starts_with("Vec")
-            {
-                quote! {
-                    seaography::TypeFilter<#ty>
-                }
-            } else {
-                let ident = format_ident!("{}EnumFilter", type_literal);
-                quote! {
-                    crate::entities::sea_orm_active_enums::#ident
-                }
-            };
+        .map(|(ident, type_ident)| {
 
             quote! {
-                #ident: Option<#filter_item>
+                #ident: Option<Box<dyn seaography::FilterTrait<Ty = #type_ident>>>
```

Comment:
Hmmmm this can't be. The GraphQL input object should be a struct / enum, it has to be something concrete.